src/AspNet.Security.OAuth.AmoCrm/AmoCrmAuthenticationDefaults.cs

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+
+namespace AspNet.Security.OAuth.AmoCrm
+{
+    /// <summary>
+    /// Default values used by the amoCRM authentication middleware.
+    /// </summary>
+    public static class AmoCrmAuthenticationDefaults
+    {
+        /// <summary>
+        /// Default value for the <see cref="Microsoft.AspNetCore.Authentication.AuthenticationScheme.Name"/>.
+        /// </summary>
+        public const string AuthenticationScheme = "amoCRM";
+
+        /// <summary>
+        /// Default value for the <see cref="Microsoft.AspNetCore.Authentication.AuthenticationScheme.DisplayName"/>.
+        /// </summary>
+        public const string DisplayName = "amoCRM";
+
+        /// <summary>
+        /// Default value for the <see cref="AuthenticationSchemeOptions.ClaimsIssuer"/>.
+        /// </summary>
+        public const string Issuer = "amoCRM";
+
+        /// <summary>
+        /// Default value for the <see cref="RemoteAuthenticationOptions.CallbackPath"/>.
+        /// </summary>
+        public const string CallbackPath = "/signin-amocrm";
+
+        /// <summary>
+        /// Default value for the <see cref="OAuthOptions.AuthorizationEndpoint"/>.
+        /// </summary>
+        public const string AuthorizationEndpoint = "https://www.amocrm.ru/oauth";
+
+        /// <summary>
+        /// Default value for the <see cref="OAuthOptions.TokenEndpoint"/>.
+        /// </summary>
+        public const string TokenEndpoint = "https://{0}.amocrm.ru/oauth2/access_token";
```

Comment:
Add `Format` suffix as this isn't a well-formed URI in of itself due to the format parameter.

See the Shopify provider for an example:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/985f785d7497f83f35f25eee8ea2de6fad7d2fc9/src/AspNet.Security.OAuth.Shopify/ShopifyAuthenticationDefaults.cs#L40

Alternatively you can use the pattern used by the Okta provider that uses paths and `IPostConfigureOptions<T>`:

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/985f785d7497f83f35f25eee8ea2de6fad7d2fc9/src/AspNet.Security.OAuth.Okta/OktaAuthenticationDefaults.cs#L45

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/985f785d7497f83f35f25eee8ea2de6fad7d2fc9/src/AspNet.Security.OAuth.Okta/OktaPostConfigureOptions.cs#L19-L31

Reply:
@martincostello, done!